packages/@aws-cdk/aws-cloudfront/lib/experimental/edge-function.ts

```diff
@@ -214,7 +214,7 @@ export class EdgeFunction extends Resource implements lambda.IVersion {
       throw new Error('stacks which use EdgeFunctions must have an explicitly set region');
     }
 
-    const edgeStackId = stackId ?? `edge-lambda-stack-${region}`;
+    const edgeStackId = stackId ?? `edge-lambda-stack-${this.stack.node.id}`;
```

Comment:
We should probably use `addr` here (an app-unique id) rather than just `id`, which can be duplicated across the app.

Reply:
I've tried using here id from `parentStack` but that has been deprecated... so I'll use `addr` instead.